app/bundles/PageBundle/Tests/Controller/PageControllerTest.php

```diff
@@ -122,4 +125,52 @@ public function LandingPageTrackingSecondVisit()
         $this->assertCount(1, $eventLogsAfterSecondVisit);
         $this->assertSame(reset($eventLogsAfterFirstVisit)['id'], reset($eventLogsAfterSecondVisit)['id']);
     }
+
+    /**
+     * Test tracking of a first visit with UTM Tags.
+     */
+    public function testLandingPageWithUtmTracking(): void
+    {
+        $timestamp  = \time();
+        $page       = $this->createTestPage();
+
+        $this->client->request('GET', "{$page->getAlias()}?utm_source=linkedin&utm_medium=social&utm_campaign=mautic&utm_content=".$timestamp);
+        $clientResponse = $this->client->getResponse();
+        // $this->assertEquals(Response::HTTP_OK, $clientResponse->getStatusCode(), 'page not found');
+        if (Response::HTTP_OK === $clientResponse->getStatusCode()) {
+            $leadModel     = $this->container->get('mautic.model.factory')->getModel('lead');
+            $allUtmTags    = $leadModel->getUtmTagRepository()->getEntities();
```

Comment:
This can be written simpler like
```suggestion
            $allUtmTags    = $this->em->getRepository(UtmTag::class)->getEntities();
```
But it basically do the same thing that you wrote, so you don't have to change that. Just FYI.